libdrgn/program.c

```diff
@@ -75,12 +92,14 @@ void drgn_program_init(struct drgn_program *prog,
 
 void drgn_program_deinit(struct drgn_program *prog)
 {
-	if (prog->prstatus_cached) {
+	if (prog->threads_cached) {
 		if (prog->flags & DRGN_PROGRAM_IS_LINUX_KERNEL)
 			drgn_prstatus_vector_deinit(&prog->prstatus_vector);
 		else
-			drgn_prstatus_map_deinit(&prog->prstatus_map);
+			drgn_thread_set_deinit(&prog->thread_set);
 	}
+	if (prog->crashed_thread)
+		drgn_thread_destroy(prog->crashed_thread);
```

Comment:
Let's make `drgn_thread_destroy()` allow (and ignore) a `NULL` `thread` instead of checking it here; that's the general convention for `destroy` functions in `drgn`.